dev/com.ibm.ws.javaee.ddmodel/test/com/ibm/ws/javaee/ddmodel/ejb/EJBJarTest.java

```diff
@@ -34,11 +34,12 @@ public void testGetVersionID() throws Exception {
         Assert.assertEquals(EJBJar.VERSION_3_0, parse(ejbJar30() + "</ejb-jar>").getVersionID());
         Assert.assertEquals(EJBJar.VERSION_3_1, parse(ejbJar31() + "</ejb-jar>").getVersionID());
         Assert.assertEquals(EJBJar.VERSION_3_2, parse(ejbJar32() + "</ejb-jar>").getVersionID());
+        Assert.assertEquals(EJBJar.VERSION_4_0, parse(ejbJar40() + "</ejb-jar>").getVersionID());
     }
 
     @Test(expected = DDParser.ParseException.class)
-    public void testMaxVersion31() throws Exception {
-        parse(ejbJar32() + "</ejb-jar>", EJBJar.VERSION_3_1);
+    public void testMaxVersion() throws Exception {
+        parse(ejbJar40() + "</ejb-jar>", EJBJar.VERSION_3_2);
```

Comment:
Is this right?  I can't tell what exactly this test is doing, but you've set DEFAULT_MAX_VERSION to 4_0 in EJBJarEntryAdapter, bumping it from 3_1.

Reply:
Apparently it is, since this test has passed in your personal build.

Reply:
this is to test ```} else if (maxVersion >= EJBJar.VERSION_4_0 && "4.0".equals(vers)) {``` gets a parse exception if the maxVersion variable is lower than 4, the test is passing in a new max version EJBJar.VERSION_3_2 so we aren't using the default

Reply:
In FATs and a live server. maxVersion is determined by what feature level is set by which ejb feature is specified in server.xml.  So, if the have an EJB 4.0 ejb-jar.xml but have enabled ejb-lite-3.2 they will get the parse exception.

Reply:
I'm not sure why we even have/had this check, I technically am introducing a behavior change now if a customer had a 3.2 ejb-jar.xml but had 3.1 or lower enabled. They will not get a parse exception anymore.

Reply:
or if there was something special about 3.2 -> 3.1 and by coincidence 3.2 was the latest and the check should still apply there and not 4.0 -> 3.2.

Reply:
Though I was wrong to get rid of the check for 3.2

Reply:
Aha.  Thanks for the clarification on what this test is doing.